raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -880,18 +878,46 @@ def run_test_batch_unlock_after_restart(raiden_network, token_addresses, deposit
         partner_address=alice_app.raiden.address,
     )
 
-    alice_app.stop()
+    # wait for the close transaction to be mined, this is necessary to compute
+    # the timeout for the settle
+    with gevent.Timeout(timeout):
+        waiting.wait_for_close(
+            raiden=alice_app.raiden,
+            payment_network_id=registry_address,
+            token_address=token_address,
+            channel_ids=[alice_bob_channel_state.identifier],
+            retry_timeout=alice_app.raiden.alarm.sleep_time,
+        )
 
-    waiting.wait_for_settle(
-        raiden=alice_app.raiden,
```

Comment:
this was actually the bug, `alice_app` was stopped above, so the test would never finish. (It did finish because `raise_on_failure` is buggy, #4003 fixes it)